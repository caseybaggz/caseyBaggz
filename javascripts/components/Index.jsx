import React from 'react';


export default class Index extends React.Component {
  constructor() {
    super();

    this.state = {
      navActive: false
    };

    // cache methods
    this._closeNav   = this._closeNav.bind(this);
    this._triggerNav = this._triggerNav.bind(this);
  }


  componentDidMount() {
    const container = document.querySelector('.index-view-container');

    window.setTimeout(() => {
      container.classList.add('fadeIn');
    }, 1000);
  }


  render() {
    return (
      <div className="index-view-container animated">
        <div className="blog-link-wrapper">
          <a href="https://medium.com/@caseybaggz" target="_blank" className="link" />
        </div>

        <span className="logo-container">
          <img className="img" src="./images/logos/baggz.svg" />
        </span>

        <div className="view-content">
          { this.props.children }
        </div>
      </div>
    );
  }


  // PRIVATE

  _closeNav() {
    this.setState({
      navActive: false
    });
  }

  _triggerNav() {
    const active = this.state.navActive;

    if (active) {
      this._closeNav();
    } else {
      this.setState({
        navActive: true
      });
    }
  }
}
