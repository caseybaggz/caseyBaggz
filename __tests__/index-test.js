import React from 'react';
import Index from '../javascripts/components/Index';
import renderer from 'react-test-renderer';


it('should render Index component successfully', () => {
  const tree = renderer.create(<Index />).toJSON();
  expect(tree).toMatchSnapshot();
});
