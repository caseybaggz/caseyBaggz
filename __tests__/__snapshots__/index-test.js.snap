exports[`test should render Index component successfully 1`] = `
<div
  className="index-view-container animated">
  <div
    className="blog-link-wrapper">
    <a
      className="link"
      href="https://medium.com/@caseybaggz"
      target="_blank" />
  </div>
  <span
    className="logo-container">
    <img
      className="img"
      src="./images/logos/baggz.svg" />
  </span>
  <div
    className="view-content" />
</div>
`;
